### PR TITLE
Hw9

### PR DESCRIPTION
**Pull request by nightdread**

### CODE REVIEW

File: hw9/hw9_vagrant_provision.sh [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+user_sudoers
+ssh_key
+update_hosts
+restrict_pass_connection
+restrict_root_access
+install_apps
+dns_update
+check_myownapp
+external_ip
+dir_owner
```
Comment (mavrik90) [inline hw9/hw9_vagrant_provision.sh]: New line

File: hw9/hw9_vagrant_provision.sh [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    sudo sh -c "echo \"$USER_NAME    ALL=(ALL:ALL) NOPASSWD:ALL\" >> /etc/sudoers" 
+}
+
+#---------GENERATE SSH KEY----------
+function ssh_key {
+    echo "Generate ssh key for: $USER_NAME"
+    sudo mkdir -p ${USER_HOME}/${USER_NAME}/.ssh
+    sudo chmod 700 ${USER_HOME}/${USER_NAME}/.ssh
+    sudo su $USER_NAME
+    ssh-keygen -t rsa -f ${USER_HOME}/${USER_NAME}/.ssh/id_rsa -q
```
Comment (mavrik90) [inline hw9/hw9_vagrant_provision.sh:38]: Please do not create a new one. Add existing key to be able to use it for login

File: hw9/Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  # Create a forwarded port mapping which allows access to a specific port
+  # within the machine from a port on the host machine. In the example below,
+  # accessing "localhost:8080" will access port 80 on the guest machine.
+  # NOTE: This will enable public access to the opened port
+  # config.vm.network "forwarded_port", guest: 80, host: 8080
+
+  # Create a forwarded port mapping which allows access to a specific port
+  # within the machine from a port on the host machine and only allow access
+  # via 127.0.0.1 to disable public access
+  # config.vm.network "forwarded_port", guest: 80, host: 8080, host_ip: "127.0.0.1"
```
Comment (mavrik90) [inline hw9/Vagrantfile:31]: Where is static IP address?

**Comment by mavrik90** [comment]
Please add pic to README.md file to proof your code.

File: hw9/hw9_vagrant_provision.sh (hunk 1 above)
Comment (nightdread) [reply hw9/hw9_vagrant_provision.sh]: complete

**Pull request merged**